### PR TITLE
docs: add property-based scheduling how-to docs

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds a how-to documentation piece for property-based scheduling.

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

N/A

### CODE REVIEW

File: docs/howtos/property-based-scheduling.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    affinity:
+        clusterAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+                clusterSelectorTerms:
+                - propertySelector:
+                    matchExpressions:
+                    - name: "kubernetes.azure.com/node-count"
+                      operator: Ge
+                      values:
+                      - 5
```
Comment (britaniar) [inline docs/howtos/property-based-scheduling.md:107]: Should be "5"

File: docs/howtos/property-based-scheduling.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                clusterSelectorTerms:
+                - labelSelector:
+                    matchLabels:
+                      region: east
+                  propertySelector:
+                    matchExpressions:
+                    - name: "kubernetes.azure.com/node-count"
+                      operator: Ge
+                      values:
+                      - 5
```
Comment (britaniar) [inline docs/howtos/property-based-scheduling.md:136]: Ditto. "5"

File: docs/howtos/affinities.md [hunk 3]
```diff
@@ -20,8 +20,16 @@ Guide.
...
+constraint that **a cluster must satisfy** before it can be picked. Each term may feature:
+
+* a label selector, which specifies a set of labels that a cluster must have or not have before 
+it can be picked;
+* a property selector, which specifies a cluster property requirement that a cluster must satisfy
+beofre it can be picked;
+* a combination of both.
+
+For the specifics about property selectors, see the
+[How-To Guide: Using Property-Based Scheduling](./property-based-scheduling.md).
```
Comment (sdesai345) [inline docs/howtos/affinities.md:32]: This link seems to be broken?

File: docs/howtos/affinities.md [hunk 4]
```diff
@@ -178,14 +186,17 @@ Each term features:
...
+* a label selector, or a property sorter.
 
 Both are required for this type of affinity terms to function.
 
 The label selector is of the same struct as the one used in
 `requiredDuringSchedulingIgnoredDuringExecution` type of affinity terms; see
 [the documentation above](#requiredduringschedulingignoredduringexecution-affinity-terms) for usage.
 
+For the specifics about property sorters, see the
+[How-To Guide: Using Property-Based Scheduling](./property-based-scheduling.md).
```
Comment (sdesai345) [inline docs/howtos/affinities.md:198]: same here (unless this links to a PR not yet merged?)

File: docs/howtos/affinities.md [hunk 5]
```diff
@@ -20,8 +20,16 @@ Guide.
...
 
 The `requiredDuringSchedulingIgnoredDuringExecution` type of affinity terms serves as a hard
-constraint that **a cluster must satisfy** before it can be picked. Each term features a label
-selector, which describes the set of labels a cluster must have or not have.
+constraint that **a cluster must satisfy** before it can be picked. Each term may feature:
+
+* a label selector, which specifies a set of labels that a cluster must have or not have before 
+it can be picked;
+* a property selector, which specifies a cluster property requirement that a cluster must satisfy
+beofre it can be picked;
```
Comment (kaysieyu) [inline docs/howtos/affinities.md:28]:
```suggestion
before it can be picked;
```

File: docs/howtos/property-based-scheduling.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+terms in the `ClusterResourcePlacement` API:
+
+* for the `requiredDuringSchedulingIgnoredDuringExecution` affinity terms, you may specify
+property selectors to filter clusters based on their properties;
+* for the `preferredDuringSchedulingIgnoredDuringExecution` affinity terms, you may specify
+property sorters to prefer clusters with a property that ranks higher or lower.
+
+# Property selectors in `requiredDuringSchedulingIgnoredDuringExecution` affinity terms
+
+A property selector is essentially an array of expression matchers against cluster properties.
```
Comment (kaysieyu) [inline docs/howtos/property-based-scheduling.md:27]:
```suggestion
A property selector is an array of expression matchers against cluster properties.
```

File: docs/howtos/property-based-scheduling.md (hunk 1 above)
Comment (michaelawyu) [reply docs/howtos/property-based-scheduling.md:107]: Done.

File: docs/howtos/property-based-scheduling.md (hunk 2 above)
Comment (michaelawyu) [reply docs/howtos/property-based-scheduling.md:136]: Done.

File: docs/howtos/affinities.md (hunk 5 above)
Comment (michaelawyu) [reply docs/howtos/affinities.md:28]: Done.

File: docs/howtos/property-based-scheduling.md (hunk 6 above)
Comment (michaelawyu) [reply docs/howtos/property-based-scheduling.md]: Done.

**Review by britaniar — APPROVED** [review]

**Pull request merged**